File: src/main/kotlin/net/rafael/web/control/network/packet/Packet.kt
package net.rafael.web.control.network.packet

import net.rafael.web.control.network.document.Document
import java.util.UUID

//------------------------------
//
// This class was developed by Rafael K.
// On 1/31/2022 at 10:30 AM
// In the project WebControl
//
//------------------------------

open class Packet(val id: Int, val document: Document) {

    val uuid: String = UUID.randomUUID().toString()

}

File: client/src/js/network/packet/Packet.ts
import * as uuid from 'uuid';

export class Packet {

    readonly id: number
    readonly uuid: string
    readonly document: {data: {}}

    constructor(id: number, data: {}) {
        this.id = id;
        this.uuid = uuid.v4();
        this.document = {data: data};
    }

}